src/qililab/calibration/calibration_node.py

```diff
@@ -0,0 +1,668 @@
+
+                # Extra parameters for this concrete notebook:
+                param1=0
+                param2=0
+                ...
+
+            |
+
+            **2) An experiment/circuit** with its corresponding loops for the sweep given by ``sweep_interval``.
+
+            .. code-block:: python
+
+                # Set the environment and paths:
+                ...
+
+                # Set the platform:
+                platform = ql.build_platform(path=platform_path)
+                platform.connect()
+                platform.initial_setup()
+                platform.turn_on_instruments()
+
+                # Define circuit
+                circuit = ...
+
+                # Loop over the sweeps executing the platform:
+                results_list = []
+                for X in sweep_interval:
+                    platform.set_parameter(alias=alias, parameter=ql.Parameter.X, value=X)
+                    result = platform.execute(program=circuit, num_avg=hw_avg, repetition_duration=repetition_duration)
+                    results_list.append(result.array)
+
+                results = np.hstack(results_list)
+
+            |
+
+            **3) An analysis procedure**, that plots and fits the obtained data to the expected theoretical behaviour and finds the optimal desired parameters.
+
+            .. code-block:: python
+
+                def fit(xdata, results):
+                    ...
+
+                fitted_values, x_data, y_data, figure = fit(xdata=sweep_interval, results=results)
+                plt.show()
+
+            |
+
+            **4) An export data cell**, that calls ``export_calibration_outputs()`` with the dictionary to retrieve from the notebook into the calibration workflow:
+
+            .. code-block:: python
+
+                from qililab.automatic_calibration.calibration_node import export_calibration_outputs
+
+                export_calibration_outputs(
+                    {
+                        "check_parameters": {"x": sweep_interval, "y": results},
+                        "platform_params": [(bus_alias0, qubit, param_name0, fitted_values[0]), (bus_alias1, qubit, param_name1, fitted_values[1])],
+                        "fidelities": [(qubit, "fidelity1", 0.9), (qubit, "fidelity2", 0.95)]  # Fidelities in the output dictionary are optional.
+                    }
+                )
+
+            where the ``check_parameters`` are a dictionary of the saved results to do comparisons against. And the ``platform_params`` are a list of parameters to set on the platform.
+    """
+
+    def __init__(
+        self,
+        nb_path: str,
+        in_spec_threshold: float,
+        bad_data_threshold: float,
+        comparison_model: Callable,
+        drift_timeout: float,
+        qubit_index: int | list[int] | None = None,
+        node_distinguisher: int | str | None = None,
+        input_parameters: dict | None = None,
+        sweep_interval: np.ndarray | None = None,
```

Comment:
I'm not convince we should need this... can't we define all sweep intervals inside the Jupyter Notebooks?

Reply:
If the intervals depend on a value (for example the frequency of the qubit) we can always get that value from the runcard

Reply:
Then we would need to define the `check_data()` interval inside the notebook as well , or a way to compute it inside.

We thought it would be clearer to have such computation in the Node class instead (`_build_check_data_interval()`)

Also this makes possible for using same notebook with different intervals, for when for example we run the first iteration of a notebook, very zoomed out, and later on a more zoomed one to calibrate better! (For example in the flipping!)

Reply:
Moving this to a new task, to close the PR 👍